rust/lance-encoding/src/encodings/physical/general.rs

```diff
@@ -134,14 +134,66 @@ impl MiniBlockDecompressor for GeneralMiniBlockDecompressor {
     }
 }
 
+/// A block compressor that wraps another block compressor and applies
+/// general-purpose compression (LZ4, Zstd) to the resulting buffer.
+#[derive(Debug)]
+pub struct GeneralBlockCompressor {
+    inner: Box<dyn BlockCompressor>,
+    compression: CompressionConfig,
+}
+
+impl GeneralBlockCompressor {
+    pub fn new(inner: Box<dyn BlockCompressor>, compression: CompressionConfig) -> Self {
+        Self { inner, compression }
+    }
+}
+
+impl BlockCompressor for GeneralBlockCompressor {
+    fn compress(&self, data: DataBlock) -> Result<LanceBuffer> {
+        let compressed = self.inner.compress(data)?;
+
+        let compressor = GeneralBufferCompressor::get_compressor(self.compression)?;
+        let mut buf = vec![];
+        compressor.compress(&compressed, &mut buf)?;
+
+        Ok(LanceBuffer::from(buf))
+    }
+}
+
+#[derive(Debug)]
+pub struct GeneralBlockDecompressor {
+    inner: Box<dyn BlockDecompressor>,
+    compression: CompressionConfig,
+}
+
+impl GeneralBlockDecompressor {
+    pub fn new(inner: Box<dyn BlockDecompressor>, compression: CompressionConfig) -> Self {
+        Self { inner, compression }
+    }
+}
+
+impl BlockDecompressor for GeneralBlockDecompressor {
+    fn decompress(&self, data: LanceBuffer, num_values: u64) -> Result<DataBlock> {
+        let decompressor = GeneralBufferCompressor::get_compressor(self.compression)?;
+        let mut buf = vec![];
+        decompressor.decompress(&data, &mut buf)?;
+
+        self.inner.decompress(LanceBuffer::from(buf), num_values)
+    }
+}
```

Comment:
Minor nit: I could go either way on this but this could probably be a single `GeneralBlock` struct that impls both traits.

Or perhaps, even better, we could just add these traits to `CompressedBufferEncoder` instead of making any new structs?